pagopa/pn-frontend
# Payments whit no attachment - pn-8869

## Short description
Sometimes the payment attachment (no F24) is included in the act and not in the payment. So when the user select a payment, the button to download the attachment must be shown/hidden based on this condition.
We also added a delay of 1 second after the selection, before the payment is enabled

## List of changes proposed in this pull request
- Updated NotificationPaymentRecipient to manage the logic described above
- Created NotificationPaymentTitle component to have a cleaner code
- Updated the tests
- Reworked the NotificationPaymentRecipient component to reduce the complexity

## How to test
- Login with pf on dev (colombo) -> search the notification AGYX-HKJK-WGTD-202311-T-1 -> select one payment and check that the button to download the attachment is shown/hidden correctly
- Login with pf on dev (ada) -> search the notification ATMG-XVDH-WPTR-202311-T-1 -> the only payment available mustn't have the button to download the attachment
- Login with pf on dev -> search a notification with only one payment and with the attachment linked to it -> he only payment available must have the button to download the attachment
